dkg-gadget/src/worker.rs

```diff
@@ -440,49 +499,93 @@ where
 		debug!(target: "dkg", "🕸️  Try sending DKG messages");
-		let authority_id = if let Some(id) =
-			self.key_store.authority_id(self.current_validator_set.authorities.as_slice())
-		{
-			debug!(target: "dkg", "🕸️  Local authority id: {:?}", id);
-			id
-		} else {
-			panic!("error");
-		};
 
-		self.rounds.proceed();
+		let send_messages = |rounds: &mut MultiPartyECDSARounds<
+			(H256, <<B as Block>::Header as Header>::Number),
+			Commitment<<<B as Block>::Header as Header>::Number, H256>,
+		>,
+		                     authority_id: Public| {
+			rounds.proceed();
+
+			if rounds.is_offline_ready() {
+				let at = BlockId::hash(self.latest_header.as_ref().unwrap().hash());
+				let pub_key = rounds
+					.get_public_key()
+					.unwrap()
+					.get_element()
+					.serialize_uncompressed()
+					.to_vec();
+				let _res = self.client.runtime_api().set_dkg_pub_key(&at, pub_key);
+			}
 
-		// TODO: run this in a different place, tied to certain number of blocks probably
-		if self.rounds.is_offline_ready() {
-			// TODO: use deterministic random signers set
-			let signer_set_id = self.current_validator_set.id;
-			let s_l = (1..=self.rounds.dkg_params().2).collect();
-			match self.rounds.reset_signers(signer_set_id, s_l) {
-				Ok(()) => info!(target: "dkg", "🕸️  Reset signers"),
-				Err(err) => error!("Error resetting signers {}", err),
+			// TODO: run this in a different place, tied to certain number of blocks probably
+			if rounds.is_offline_ready() {
+				// TODO: use deterministic random signers set
+				let signer_set_id = self.current_validator_set.id;
+				let s_l = (1..=rounds.dkg_params().2).collect();
+				match rounds.reset_signers(signer_set_id, s_l) {
+					Ok(()) => info!(target: "dkg", "🕸️  Reset signers"),
+					Err(err) => error!("Error resetting signers {}", err),
+				}
 			}
-		}
 
-		for message in self.rounds.get_outgoing_messages() {
-			let dkg_message = DKGMessage { id: authority_id.clone(), payload: message };
-			let encoded_dkg_message = dkg_message.encode();
-			debug!(
-				target: "dkg",
-				"🕸️  DKG Message: {:?}, encoded: {:?}",
-				dkg_message,
-				encoded_dkg_message
-			);
+			for message in rounds.get_outgoing_messages() {
+				let dkg_message = DKGMessage { id: authority_id.clone(), payload: message };
+				let encoded_dkg_message = dkg_message.encode();
+				debug!(
+					target: "dkg",
+					"🕸️  DKG Message: {:?}, encoded: {:?}",
+					dkg_message,
+					encoded_dkg_message
+				);
 
-			self.gossip_engine.lock().gossip_message(
-				dkg_topic::<B>(),
-				encoded_dkg_message.clone(),
-				true,
-			);
-			trace!(target: "dkg", "🕸️  Sent DKG Message {:?}", encoded_dkg_message);
+				self.gossip_engine.lock().gossip_message(
+					dkg_topic::<B>(),
+					encoded_dkg_message.clone(),
+					true,
+				);
+				trace!(target: "dkg", "🕸️  Sent DKG Message {:?}", encoded_dkg_message);
+			}
+		};
+
+		if self.current_validator_set.id == GENESIS_AUTHORITY_SET_ID {
```

Comment:
We should probably change this condition, since current dkg rounds need always send their outgoing messages for voting process. Maybe something like this?
`if let Some(id) =
			self.key_store.authority_id(self.current_validator_set.authorities.as_slice())
		{
			debug!(target: "dkg", "🕸️  Local authority id: {:?}", id);
			send_messages(&mut self.rounds, id);
		} else {
			panic!("error");
		}
		if self.current_validator_set.id != GENESIS_AUTHORITY_SET_ID {
			if let Some(id) =
				self.key_store.authority_id(self.queued_validator_set.authorities.as_slice())
			{
				debug!(target: "dkg", "🕸️  Local authority id: {:?}", id);
				if let Some(next_rounds) = self.next_rounds.as_mut() {
					send_messages(next_rounds, id);
				}
			} else {
				panic!("error");
			}
		}`